src/compat.h

```diff
@@ -102,12 +102,12 @@ typedef void* sockopt_arg_type;
 typedef char* sockopt_arg_type;
 #endif
 
+#if not defined WIN32
```

Comment:
nit: MSVC does not like this. I would suggest to use `#ifndef`

Reply:
I think the official syntax, which works with any C++ compiler, is
```c++
#if !defined(WIN32)
```